Plugins/Creature/NWScript/nwnx_creature.nss

```diff
@@ -310,6 +310,15 @@ void NWNX_Creature_SetFeatRemainingUses(object creature, int feat, int uses);
 int NWNX_Creature_GetTotalEffectBonus(object creature, int bonusType=NWNX_CREATURE_BONUS_TYPE_ATTACK, object target=OBJECT_INVALID, int isElemental=0,
     int isForceMax=0, int savetype=-1, int saveSpecificType=-1, int skill=-1, int abilityScore=-1, int isOffhand=FALSE);
 
+// Set the original first or last name of creature
+//
+// For PCs this will persist to the .bic file if saved. Requires a relog to update.
+void NWNX_Creature_SetOriginalName(object creature, string name, int isLastName);
+
+// Get the original first or last name of creature
+string NWNX_Creature_GetOriginalName(object creature, int isLastName);
```

Comment:
Getter already provided in the base game:
```C
// Set the name of oObject.
//
// - oObject: the object for which you are changing the name (area, creature, placeable, item, or door).
// - sNewName: the new name that the object will use.
// Note: SetName() does not work on player objects.
//       Setting an object's name to "" will make the object
//       revert to using the name it had originally before any
//       SetName() calls were made on the object.
string GetName(object oObject, int bOriginalName=FALSE);
```

Reply:
True, but it returns the full name so you can't differentiate between the first or last name.

Reply:
Yep, true. Both of those could have spaces. Let's leave it in. :)